google-http-client/src/main/java/com/google/api/client/http/HttpResponseException.java

```diff
@@ -253,31 +292,45 @@ public HttpHeaders getHeaders() {
     /**
      * Sets the HTTP response headers.
      *
-     * <p>
-     * Overriding is only supported for the purpose of calling the super implementation and changing
-     * the return type, but nothing else.
-     * </p>
+     * <p>Overriding is only supported for the purpose of calling the super implementation and
+     * changing the return type, but nothing else.
      */
     public Builder setHeaders(HttpHeaders headers) {
       this.headers = Preconditions.checkNotNull(headers);
       return this;
     }
 
     /** Returns the HTTP response content or {@code null} for none. */
-    public final String getContent() {
-      return content;
+    public final String getContent() throws IOException {
+      if (content == null) {
+        return null;
+      }
+      return new String(content, charset.name());
     }
 
     /**
      * Sets the HTTP response content or {@code null} for none.
      *
-     * <p>
-     * Overriding is only supported for the purpose of calling the super implementation and changing
-     * the return type, but nothing else.
-     * </p>
+     * <p>Overriding is only supported for the purpose of calling the super implementation and
+     * changing the return type, but nothing else.
      */
     public Builder setContent(String content) {
-      this.content = content;
+      if (content != null) {
+        this.content = content.getBytes();
+      }
+      return this;
+    }
+
+    /**
+     * Sets the HTTP response charset or {@code null} for none.
+     *
+     * <p>Overriding is only supported for the purpose of calling the super implementation and
+     * changing the return type, but nothing else.
+     */
+    public Builder setCharset(Charset charset) {
```

Comment:
Do we need this to be public?

Reply:
@sduskis, the public access to this method used in HttpResponseExceptionTest class for unit tests.

Reply:
If this is used only in tests, can we mark this package private (i.e. no public/protected/private modifier), and add the `@VisibleForTesting` annotation?

Reply:
Then non-public and @VisibleForTesting